config/routes.rb

```diff
@@ -40,7 +40,7 @@ def matches?(request)
   resource :affordability, path: "cyfrifiannell-fforddiadwyedd-morgais", as: 'affordability_cy' do
     get '/', to: "affordabilities#step_1"
     collection do
-      get 'step_1', path: "step-1"
+      match 'step_1', path: "step-1", via: [:get, :post]
```

Comment:
@KateWilkinson I feel that this should have a regression test. The fact that this bug was in there and it didn't fail any tests raises concerns. I believe that this newly-added route code should be tested.

```ruby
#file: config/routes.rb
    match 'step_1', path: "step-1", via: [:get, :post]

#file: spec/routings/affordability_steps_spec.rb

   1 require 'spec_helper'
   2
   3 RSpec.describe MortgageCalculator::AffordabilitiesController, type: :routing do
   4   describe 'GET /step_1' do
   5     it 'routes to first step in English' do
   6       expect(get '/mortgage-affordability-calculator/step-1').to be_routable
   7     end
   8   end
   9
  10   describe 'GET /cyfrifiannell-fforddiadwyedd-morgais/step-1' do
  11     it 'routes to first step in Welsh' do
  12       expect(get '/cyfrifiannell-fforddiadwyedd-morgais/step-1').to be_routable
  13     end
  14   end
  15 end
```

I have tried these tests. They are not currently passing. It's just an illustration of the kind of retrospective test I want to see. Just so that we are assured that if the route is broken the tests will fail, thereby raising the alarm.

Wdyt @KateWilkinson ?

Reply:
Hi @Tr4pSt3R, I totally agree with you it would have been good to have tests there that caught this bug. My only concern is the delay in getting these fixes out into production, but if it's ok to wait then yes, agree it'd be good to add tests around this.

Reply:
@KateWilkinson try this spec file below. The problem with this test, however, is that it still passes for the old code which you remove. So.. maybe you can improve on my test :) to ensure it catches the error

```
# file: spec/routings/affordability_steps_spec.rb

   1 require 'spec_helper'
   2
   3 RSpec.describe MortgageCalculator::AffordabilitiesController, type: :routing do
   4   routes {MortgageCalculator::Engine.routes }
   5
   6   describe 'GET /step_1' do
   7     it 'routes to first step' do
   8       expect(get '/mortgage-affordability-calculator/step-1').to be_routable
   9     end
  10   end
  11
  12   describe 'GET /cyfrifiannell-fforddiadwyedd-morgais/step-1' do
  13     it 'routes to first step in Welsh' do
  14       expect(get '/cyfrifiannell-fforddiadwyedd-morgais/step-1').to be_routable
  15     end
  16   end
  17 end
```

Reply:
Hey @Tr4pSt3R could you have a look at the spec I've added and see what you think?